tests/attr/test_targets.py

```diff
@@ -1,668 +1,205 @@
+    list or tensor.
+    Attribution of each example is computed independently with the appropriate target
+    and compared to the corresponding result of attributing to a batch with a tensor
+    / list of targets.
+    """
+
+    def __new__(cls, name: str, bases: Tuple, attrs: Dict):
+        for test_config in config:
+            algorithms = cast(List[Type[Attribution]], test_config["algorithms"])
+            model = test_config["model"]
+            args = cast(Dict[str, Any], test_config["attribute_args"])
+            layer = test_config["layer"] if "layer" in test_config else None
+            target_delta = (
+                test_config["target_delta"] if "target_delta" in test_config else 0.0001
+            )
+            noise_tunnel = (
+                test_config["noise_tunnel"] if "noise_tunnel" in test_config else False
+            )
+            baseline_distr = (
+                test_config["baseline_distr"]
+                if "baseline_distr" in test_config
+                else False
+            )
 
-
-class Test(BaseTest):
-    def test_simple_target_missing_error(self):
+            if "target" not in args or not isinstance(args["target"], (list, Tensor)):
+                continue
+
+            for algorithm in algorithms:
+                # FeaturePermutation requires a batch of inputs
+                # so skipping tests
+                if issubclass(algorithm, FeaturePermutation):
+                    continue
+                test_method = cls.make_single_target_test(
+                    algorithm,
+                    model,
+                    layer,
+                    args,
+                    target_delta,
+                    noise_tunnel,
+                    baseline_distr,
+                )
+                test_name = (
+                    "test_target_"
+                    + cast(str, test_config["name"])
+                    + "_"
+                    + algorithm.__name__
+                    + ("NoiseTunnel" if noise_tunnel else "")
+                )
+                if test_name in attrs:
+                    raise AssertionError(
+                        "Trying to overwrite existing test with name: %r" % test_name
+                    )
+                attrs[test_name] = test_method
+        return super(TargetsMeta, cls).__new__(cls, name, bases, attrs)
+
+    @classmethod
+    @deep_copy_args
+    def make_single_target_test(
+        cls,
+        algorithm: Type[Attribution],
+        model: Module,
+        layer: Module,
+        args: Dict[str, Any],
+        target_delta: float,
+        noise_tunnel: bool,
+        baseline_distr: bool,
+    ) -> Callable:
+        """
+        This method creates a single target test for the given algorithm and parameters.
+        """
+
+        target_layer = get_nested_attr(model, layer) if layer is not None else None
+        original_inputs = args["inputs"]
```

Comment:
It looks like that we didn't need to copy the inputs for the data parallel and for the jit but we need it for targets. Maybe you can explain a little bit.

Reply:
Targets needs to change the inputs / additional args etc. to test each example independently, will add a comment to make it clearer.